### ISSUE TITLE
[3.3.5][ 6.x] DB/Creature: Trained Rock Falcon doesn't fly

### ISSUE THREAD

**Issue by sirikfoll** [TrinityCore/TrinityCore#16191]
Npc Trained Rock Falcon (http://wotlk.openwow.com/npc=24783)
doesn't fly, regardless of which InhabitType is set(current is 4)

He is needed for quest [There Exists No Honor Among Birds](http://wotlk.openwow.com/quest=11470)

Probably is a missing aura, don't have data to discover if is.

hash 893c63f32039a9d5bf7c225ac918297d9bdc375a
branch 3.3.5

**Comment by dr-j** [comment]
Actually I checked this quest not long ago due to reports of it not working found a couple of things

You can complete these quest by using ability on trained rock falcon without even been anywhere near any fjord hawks or fjord hawk eggs.

**Comment by Saben65** [comment]
Video guide of the quest in retail for reference: https://www.youtube.com/watch?v=EpqOQFo-4Fw

**Comment by sirikfoll** [comment]
The Spell Scavenge(44422)  has
```
Effect 0: Id 86 (SPELL_EFFECT_ACTIVATE_OBJECT)
BasePoints = 1
Targets (40, 0) (TARGET_GAMEOBJECT_NEARBY_ENTRY, NO_TARGET)
EffectMiscValueA = 15
```
but this condition doesn't seem to work, the GO isn't activated
```sql
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND `SourceEntry`=44422;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
(13,1,44422,0,0,31,0,5,186946,0,0,0,0,'',"Spell 'Scavenge' target GameObject 'Fjord Hawk Egg'"),
```

to prevent using the ability anywhere, it needs a core script like this?
```cpp
SpellCastResult CheckCast()
{
    GameObject* go = GetCaster()->FindNearestGameObject(186946, 3.0f);
    if (!go || go->GetRespawnTime())
         return SPELL_FAILED_REQUIRES_SPELL_FOCUS;
    return SPELL_CAST_OK;
}

**Comment by Killyana** [comment]
SourceType 17 doesn't fix the issue?

**Comment by Treeston** [comment]
Confirming 3.3.5 d1a185e3d5af

**Comment by Rushor** [comment]
> SourceType 17 doesn't fix the issue?

it does, flying shit can be fixed with flying aura in creature_template_addon

**Comment by tkrokli** [comment]
http://www.wowhead.com/npc=24783/trained-rock-falcon
```sql
-- add aura 'Flying' to Trained Rock Falcon
DELETE FROM `creature_template_addon` WHERE `entry`= 24783;
INSERT INTO `creature_template_addon` (`entry`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES
(24783, 0, 0, 0, 1, 0, 54422);

DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`= 17 AND `SourceEntry`= 44422;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
(17,0,44422,0,0,31,0,5,186946,0,0,0,0,'',"Spell 'Scavenge' target GameObject 'Fjord Hawk Egg'");
```
I am able to control the Trained Rock Falcon and access the 3 spells in the button bar (1,2,3) as well as the hotbar for cancelling the channeled spell when needed. I am able to fly down to the eggs, but every time I click button 1 (Scavenge, spell 44422), I get an error message in red text: **You can't do that yet**. The 2 other spells, Soar (http://www.wowhead.com/spell=44423/soar) and Escape (http://www.wowhead.com/spell=44424/escape) work as intended, at least when targeting "myself".

The only way I was able to trigger a cooldown on `Scavenge` (1) was when I pressed F1 two times to select my falcon vehicle, but only after triggering the cooldown of Soar (2) and Escape (3).

I did get some error messages in Worldserver console about spell 44408 (http://www.wowhead.com/spell=44408/hawk-hunting):
```
Spell::EffectForceCast of spell 44408: triggering unknown spell id 0
```
but those were because I clicked the falcon hunting spell icon (http://www.wowhead.com/spell=44420/hawk-hunting) while already controlling the falcon. I just mention this for a complete feedback.

I also get this weird message while trying to use the normal spell buttons:
```
WorldObject::MovePositionToFirstCollision invalid coordinates X: 18901.712891 and Y: -52474.855469 were passed!
```
I am starting to wonder if the `conditions` script should use `CONDITION_NEAR_GAMEOBJECT` (30) instead of `CONDITION_OBJECT_ENTRY_GUID` (31) . Does that make sense, or is it something I should know about the difference between those two `ConditionTypeOrReference` values?

**Comment by Treeston** [comment]
Still issue as of 3.3.5 f37682b

@tkrokli implicit target condition on scavenge spell to select GO, I think.

**Comment by dr-j** [comment]
implicit target does not prevent a spell from being cast just if condition is not met effects wont hit anything

conditiontype 17 does prevent spell from being cast if target is not met

**Comment by Treeston** [comment]
Yes, but implicit target will make it auto-select the nearby GO. You need both conditions.

**Comment by tkrokli** [comment]
Interesting points from both of you, I hope to find some time to do more testing during Sunday. I presume there will have to be both a `CONDITION_SOURCE_TYPE_SPELL_IMPLICIT_TARGET (13)` and `CONDITION_SOURCE_TYPE_SPELL = 17`

**Comment by Kodekc** [comment]
Perhaps useful
[21463_2016-04-14_23-33-45_02E2C59E_parsed.txt](https://github.com/TrinityCore/TrinityCore/files/219961/21463_2016-04-14_23-33-45_02E2C59E_parsed.txt)

**Comment by Keader** [comment]
Merged with: #16921
